### ISSUE TITLE
Init SPI bus in main and just the SPI device on TFT and touch controller 

### ISSUE THREAD

**Issue by C47D** [littlevgl/lv_port_esp32#50]
Hi,

Right now we init the SPI bus and SPI devices on the TFT and touch init functions, like so:

https://github.com/littlevgl/lv_port_esp32_ili9341/blob/6c0a081029bcb05beb01033d6f12eb580ab999e8/components/lvgl_ili9341/disp_spi.c#L47-L82

I think is better to init the SPI bus on main so the user can add more devices, like sensors, etc, on main instead of modifying the TFT and touch controller functions to do so, maybe something like this:

In disp_spi.c:
```c
void disp_spi_init(spi_host_device_thost host)
{
    esp_err_t ret;

    spi_device_interface_config_t devcfg={
            .clock_speed_hz=40*1000*1000,           //Clock out at 40 MHz
            .mode=0,                                //SPI mode 0
            .spics_io_num=DISP_SPI_CS,              //CS pin
            .queue_size=1,
            .pre_cb=NULL,
            .post_cb=spi_ready,
            .flags = SPI_DEVICE_HALFDUPLEX
    };

    //Attach the LCD to the SPI bus
    ret=spi_bus_add_device(host, &devcfg, &spi);
    assert(ret==ESP_OK);
}
```

In main.c:
```c
void app_main()
{
	lv_init();

    spi_bus_config_t buscfg={
            .miso_io_num=-1,
            .mosi_io_num=DISP_SPI_MOSI,
            .sclk_io_num=DISP_SPI_CLK,
            .quadwp_io_num=-1,
            .quadhd_io_num=-1,
#if CONFIG_LVGL_TFT_DISPLAY_CONTROLLER == 0
            .max_transfer_sz = DISP_BUF_SIZE * 2,
#elif CONFIG_LVGL_TFT_DISPLAY_CONTROLLER == 1
            .max_transfer_sz = DISP_BUF_SIZE * 3,
#endif
    };

    //Initialize the SPI bus
    ret=spi_bus_initialize(host, &buscfg, 1);
    assert(ret==ESP_OK);

	disp_spi_init(HSPI_HOST);
	disp_driver_init();

#if ENABLE_TOUCH_INPUT
	tp_spi_init();
	xpt2046_init();
#endif

    static lv_color_t buf1[DISP_BUF_SIZE];
    static lv_color_t buf2[DISP_BUF_SIZE];
    static lv_disp_buf_t disp_buf;
    lv_disp_buf_init(&disp_buf, buf1, buf2, DISP_BUF_SIZE);

	lv_disp_drv_t disp_drv;
	lv_disp_drv_init(&disp_drv);
	disp_drv.flush_cb = disp_driver_flush;
	disp_drv.buffer = &disp_buf;
	lv_disp_drv_register(&disp_drv);

#if ENABLE_TOUCH_INPUT
    lv_indev_drv_t indev_drv;
    lv_indev_drv_init(&indev_drv);
    indev_drv.read_cb = xpt2046_read;
    indev_drv.type = LV_INDEV_TYPE_POINTER;
    lv_indev_drv_register(&indev_drv);
#endif

	esp_register_freertos_tick_hook(lv_tick_task);

	demo_create();

	while(1) {
		vTaskDelay(1);
		lv_task_handler();
	}
}
```

**Comment by kisvegabor** [comment]
> I think is better to init the SPI bus on main so the user can add more devices, like sensors, etc, on main instead of modifying the TFT and touch controller functions to do so, maybe something like this:

I agree. It'd be really clearer this way.

**Comment by C47D** [comment]
Great, I will do some tests to implement it. I think @jeremyjh have something similar in his repo, any suggestions?

**Comment by C47D** [comment]
Closed with #55
